src/Form/EventRegistrationType.php

```diff
@@ -9,21 +9,41 @@
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
 
 class EventRegistrationType extends AbstractType
 {
+    private $authorizationChecker;
+
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+    }
+
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
             ->add('firstName', TextType::class, [
                 'filter_emojis' => true,
             ])
+            ->add('lastName', TextType::class)
             ->add('emailAddress', EmailType::class)
-            ->add('postalCode', TextType::class)
-            ->add('newsletterSubscriber', CheckboxType::class, [
-                'required' => false,
+            ->add('acceptTerms', CheckboxType::class, [
+                'mapped' => false,
             ])
         ;
+
+        /** @var EventRegistrationCommand $registration */
+        $registration = $builder->getData();
+        if ($builder->getData() instanceof EventRegistrationCommand
+            && (!$this->authorizationChecker->isGranted('IS_AUTHENTICATED_FULLY')
```

Comment:
```
If (!$this->authorizationChecker(...) || ($resgistration instanceof ... && !$registration->is...)) {
```

Reply:
Changed in this PR https://github.com/EnMarche/en-marche.fr/pull/2024/files#diff-44eb87946c3878749aba2837e96ea962R38